{
	"alias": "FEDORA-2020-fe82e2a834",
	"autokarma": true,
	"autotime": true,
	"bugs": [
		{
			"bug_id": 1854926,
			"feedback": [
				{
					"bug_id": 1854926,
					"comment": {
						"id": 1465919,
						"karma": 1,
						"text": "Works for me.",
						"timestamp": "2020-07-22 05:17:21",
						"update_id": 220515,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/bef150da9235aef1b7225ecc5f758fba35fa9d3f1711c31f5d1754fc6d2bf3ff?s=24&d=retro",
							"email": "ericbasil.rabi@gmail.com",
							"groups": [
								{
									"name": "ipausers"
								},
								{
									"name": "signed_fpca"
								}
							],
							"id": 3650,
							"name": "basilrabi",
							"openid": "basilrabi.id.fedoraproject.org"
						},
						"user_id": 3650
					},
					"comment_id": 1465919
				},
				{
					"bug_id": 1854926,
					"comment": {
						"id": 1466062,
						"karma": 1,
						"timestamp": "2020-07-22 09:03:07",
						"update_id": 220515,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/b2ae13be7cf78c4fa85d027daac07c34a0ef754116514c10ad0ae13f7d7cbfec?s=24&d=retro",
							"email": "ego.cordatus@gmail.com",
							"groups": [
								{
									"name": "packager"
								},
								{
									"name": "ipausers"
								},
								{
									"name": "fedora-contributor"
								},
								{
									"name": "signed_fpca"
								},
								{
									"name": "fedorabugs"
								},
								{
									"name": "rust-sig"
								}
							],
							"id": 4120,
							"name": "atim",
							"openid": "atim.id.fedoraproject.org"
						},
						"user_id": 4120
					},
					"comment_id": 1466062
				},
				{
					"bug_id": 1854926,
					"comment": {
						"id": 1466198,
						"karma": 1,
						"text": "Works for me. No regressions noted compared to previous version.",
						"timestamp": "2020-07-22 10:37:07",
						"update_id": 220515,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/d2961abadbc61ac02ca743308c188aa38757b1751e713948c0e71ea034847140?s=24&d=retro",
							"email": "chr77@protonmail.com",
							"groups": [
								{
									"name": "packager"
								},
								{
									"name": "ipausers"
								},
								{
									"name": "fedora-contributor"
								},
								{
									"name": "signed_fpca"
								},
								{
									"name": "fedorabugs"
								}
							],
							"id": 308,
							"name": "chr77",
							"openid": "chr77.id.fedoraproject.org"
						},
						"user_id": 308
					},
					"comment_id": 1466198
				},
				{
					"bug_id": 1854926,
					"comment": {
						"id": 1503858,
						"karma": 1,
						"timestamp": "2020-07-23 16:50:53",
						"update_id": 220515,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/4d5bf123c258c90abbf2b56c3da4752c620faae0f5ecb9906599f47b4ea52d84?s=24&d=retro",
							"email": "oguz@ersen.moe",
							"groups": [
								{
									"name": "qa"
								},
								{
									"name": "packager"
								},
								{
									"name": "ipausers"
								},
								{
									"name": "fedora-contributor"
								},
								{
									"name": "signed_fpca"
								},
								{
									"name": "fedorabugs"
								},
								{
									"name": "l10n"
								},
								{
									"name": "fedora-join"
								},
								{
									"name": "ask-fedora"
								}
							],
							"id": 5615,
							"name": "ersen",
							"openid": "ersen.id.fedoraproject.org"
						},
						"user_id": 5615
					},
					"comment_id": 1503858
				},
				{
					"bug_id": 1854926,
					"comment": {
						"id": 1505989,
						"karma": 1,
						"text": "working fine",
						"timestamp": "2020-07-24 10:52:56",
						"update_id": 220515,
						"user": {
							"avatar": "https://seccdn.libravatar.org/avatar/f8db9b35a53624d807b2dd11c2109d12585c5852f0ff616077f8ed9348854fb1?s=24&d=retro",
							"email": "marttikuosmanen@yahoo.co.uk",
							"groups": [
								{
									"name": "ipausers"
								},
								{
									"name": "signed_fpca"
								}
							],
							"id": 417,
							"name": "kuosmanen",
							"openid": "kuosmanen.id.fedoraproject.org"
						},
						"user_id": 417
					},
					"comment_id": 1505989
				}
			],
			"parent": true,
			"security": true,
			"title": "CVE-2020-14422 python: DoS via inefficiency in IPv{4,6}Interface classes",
			"bugzilla": {
				"bug_id": "1854926",
				"alias": "CVE-2020-14422",
				"creation_ts": "2020-07-08 12:52:24 +0000",
				"short_desc": "CVE-2020-14422 python: DoS via inefficiency in IPv{4,6}Interface classes",
				"delta_ts": "2021-02-16 19:42:16 +0000",
				"bug_status": "CLOSED",
				"resolution": "ERRATA",
				"keywords": "Security",
				"priority": "medium",
				"bug_severity": "medium",
				"depends_on": [
					"1854930",
					"1854931",
					"1854932",
					"1854934",
					"1854936",
					"1854937",
					"1854938",
					"1854939",
					"1854940",
					"1854941",
					"1854942",
					"1854943",
					"1856382",
					"1856383",
					"1856384",
					"1856385",
					"1856386",
					"1857276",
					"1857277",
					"1857278",
					"1857279",
					"1857280",
					"1857282",
					"1857283",
					"1857284",
					"1857285",
					"1857286",
					"1857287",
					"1857288",
					"1857289",
					"1857292",
					"1857293",
					"1857294",
					"1858216",
					"1885290"
				],
				"blocked": [
					{
						"bug_id": "1854944",
						"error": "NotPermitted"
					},
					{
						"bug_id": "1854945",
						"error": "NotPermitted"
					}
				],
				"external_bugs": {
					"text": "RHSA-2020:5010",
					"name": "Red Hat Product Errata"
				},
				"long_desc": [
					{
						"isprivate": "0",
						"commentid": "14088166",
						"comment_count": "0",
						"who": {
							"text": "darunesh",
							"name": "Dhananjay Arunesh"
						},
						"bug_when": "2020-07-08 12:52:24 +0000",
						"thetext": "A vulnerability was found in Lib/ipaddress.py in Python through 3.8.3 improperly computes hash values in the IPv4Interface and IPv6Interface classes, which might allow a remote attacker to cause a denial of service if an application is affected by the performance of a dictionary containing IPv4Interface or IPv6Interface objects, and this attacker can cause many dictionary entries to be created.\n\nReferences:\nhttps://bugs.python.org/issue41004\nhttps://github.com/python/cpython/pull/20956"
					},
					{
						"isprivate": "0",
						"commentid": "14088216",
						"comment_count": "1",
						"who": {
							"text": "darunesh",
							"name": "Dhananjay Arunesh"
						},
						"bug_when": "2020-07-08 12:56:50 +0000",
						"thetext": "Created mingw-python3 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854936]\n\n\nCreated python2 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854931]\n\n\nCreated python26 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854934]\n\n\nCreated python27 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854937]\n\n\nCreated python3 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854932]\n\n\nCreated python34 tracking bugs for this issue:\n\nAffects: epel-all [bug 1854930]\nAffects: fedora-all [bug 1854938]\n\n\nCreated python35 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854939]\n\n\nCreated python36 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854940]\n\n\nCreated python37 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854941]\n\n\nCreated python38 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854942]\n\n\nCreated python39 tracking bugs for this issue:\n\nAffects: fedora-all [bug 1854943]"
					},
					{
						"isprivate": "0",
						"commentid": "14111754",
						"comment_count": "4",
						"who": {
							"text": "rschiron",
							"name": "Riccardo Schirone"
						},
						"bug_when": "2020-07-15 15:12:38 +0000",
						"thetext": "The hash value returned by IPv4Interface/IPv6Interface classes is wrongly a constant value. When an IPvXInterface object is used as key of a python dictionary, the hash value is used to determine in which hash table bucket the object needs to be put. For a large number of elements, the constant hash value transforms most dictionary operation in O(n) instead of the expected O(1), making a program much slower."
					},
					{
						"isprivate": "0",
						"commentid": "14111763",
						"comment_count": "5",
						"who": {
							"text": "rschiron",
							"name": "Riccardo Schirone"
						},
						"bug_when": "2020-07-15 15:14:01 +0000",
						"thetext": "python3 embeds the ipaddress module, where the flaw lies. Instead python2 does not embeds it, but a separate python-ipaddress package is provided. Moreover, the ipaddress module is usually embedded in python-pip as well."
					},
					{
						"isprivate": "0",
						"commentid": "14111789",
						"comment_count": "6",
						"who": {
							"text": "rschiron",
							"name": "Riccardo Schirone"
						},
						"bug_when": "2020-07-15 15:20:37 +0000",
						"thetext": "Statement:\n\nIn Red Hat Enterprise Linux, python3 includes the ipaddress module by default, while for python2 a separate package, python-ipaddress, needs to be installed for the module to be used. Moreover, the ipaddress module is included in other packages as well, like python-pip."
					},
					{
						"isprivate": "0",
						"commentid": "14111860",
						"comment_count": "8",
						"who": {
							"text": "rschiron",
							"name": "Riccardo Schirone"
						},
						"bug_when": "2020-07-15 15:31:17 +0000",
						"thetext": "Created python-ipaddress tracking bugs for this issue:\n\nAffects: fedora-all [bug 1857292]\n\n\nCreated python-pip tracking bugs for this issue:\n\nAffects: epel-all [bug 1857294]\nAffects: fedora-all [bug 1857293]"
					},
					{
						"isprivate": "0",
						"commentid": "14111949",
						"comment_count": "9",
						"who": {
							"text": "mhroncok",
							"name": "Miro Hronƒçok"
						},
						"bug_when": "2020-07-15 15:59:30 +0000",
						"thetext": "Pip does not have dictionaries containing IPv4Interface or IPv6Interface objects. I don't think it's worth patching the bundled ipaddress module in it."
					},
					{
						"isprivate": "0",
						"commentid": "14118275",
						"comment_count": "12",
						"who": {
							"text": "rschiron",
							"name": "Riccardo Schirone"
						},
						"bug_when": "2020-07-17 09:21:58 +0000",
						"thetext": "Mitigation:\n\nAs a short term solution, if your application is using the IPv4Interface/IPv6Interface classes as keys of a dictionary, it is possible to patch the __hash__ method of those classes to not be constant.\n```\nIPv4Interface.__hash__ = lambda self: hash((self._ip, self._prefixlen, int(self.network.network_address)))\nIPv6Interface.__hash__ = lambda self: hash((self._ip, self._prefixlen, int(self.network.network_address)))\n```"
					},
					{
						"isprivate": "0",
						"commentid": "14405951",
						"comment_count": "18",
						"who": {
							"text": "errata-xmlrpc",
							"name": "errata-xmlrpc"
						},
						"bug_when": "2020-10-19 18:05:40 +0000",
						"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Software Collections for Red Hat Enterprise Linux 6\n  Red Hat Software Collections for Red Hat Enterprise Linux 7\n  Red Hat Software Collections for Red Hat Enterprise Linux 7.6 EUS\n  Red Hat Software Collections for Red Hat Enterprise Linux 7.7 EUS\n\nVia RHSA-2020:4285 https://access.redhat.com/errata/RHSA-2020:4285"
					},
					{
						"isprivate": "0",
						"commentid": "14406372",
						"comment_count": "19",
						"who": {
							"text": "prodsec-dev",
							"name": "Product Security DevOps Team"
						},
						"bug_when": "2020-10-19 20:21:15 +0000",
						"thetext": "This bug is now closed. Further updates for individual products will be reflected on the CVE page(s):\n\nhttps://access.redhat.com/security/cve/cve-2020-14422"
					},
					{
						"isprivate": "0",
						"commentid": "14410057",
						"comment_count": "22",
						"who": {
							"text": "errata-xmlrpc",
							"name": "errata-xmlrpc"
						},
						"bug_when": "2020-10-20 20:00:07 +0000",
						"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Software Collections for Red Hat Enterprise Linux 7\n  Red Hat Software Collections for Red Hat Enterprise Linux 7.6 EUS\n  Red Hat Software Collections for Red Hat Enterprise Linux 7.7 EUS\n\nVia RHSA-2020:4299 https://access.redhat.com/errata/RHSA-2020:4299"
					},
					{
						"isprivate": "0",
						"commentid": "14468339",
						"comment_count": "23",
						"who": {
							"text": "errata-xmlrpc",
							"name": "errata-xmlrpc"
						},
						"bug_when": "2020-11-04 00:51:35 +0000",
						"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8\n\nVia RHSA-2020:4433 https://access.redhat.com/errata/RHSA-2020:4433"
					},
					{
						"isprivate": "0",
						"commentid": "14471264",
						"comment_count": "24",
						"who": {
							"text": "errata-xmlrpc",
							"name": "errata-xmlrpc"
						},
						"bug_when": "2020-11-04 02:35:56 +0000",
						"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 8\n\nVia RHSA-2020:4641 https://access.redhat.com/errata/RHSA-2020:4641"
					},
					{
						"isprivate": "0",
						"commentid": "14492956",
						"comment_count": "26",
						"who": {
							"text": "errata-xmlrpc",
							"name": "errata-xmlrpc"
						},
						"bug_when": "2020-11-10 12:59:24 +0000",
						"thetext": "This issue has been addressed in the following products:\n\n  Red Hat Enterprise Linux 7\n\nVia RHSA-2020:5010 https://access.redhat.com/errata/RHSA-2020:5010"
					}
				]
			}
		}
	],
	"builds": [
		{
			"epoch": 0,
			"nvr": "python38-3.8.4-1.fc31",
			"release_id": 33,
			"signed": true,
			"type": "rpm",
			"package": {
				"aarch64": [
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "aarch64"
					},
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "aarch64"
					},
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "aarch64"
					}
				],
				"armv7hl": [
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "armv7hl"
					},
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "armv7hl"
					},
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "armv7hl"
					}
				],
				"i686": [
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "i686"
					},
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "i686"
					},
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "i686"
					}
				],
				"ppc64le": [
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "ppc64le"
					},
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "ppc64le"
					},
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "ppc64le"
					}
				],
				"s390x": [
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "s390x"
					},
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "s390x"
					},
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "s390x"
					}
				],
				"src": [
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "src"
					}
				],
				"x86_64": [
					{
						"name": "python38-debugsource",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "x86_64"
					},
					{
						"name": "python38-debuginfo",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "x86_64"
					},
					{
						"name": "python38",
						"version": "3.8.4",
						"release": "1.fc31",
						"arch": "x86_64"
					}
				]
			}
		}
	],
	"close_bugs": true,
	"comments": [
		{
			"id": 1458760,
			"text": "This update has been submitted for testing by thrnciar. ",
			"timestamp": "2020-07-15 12:33:26",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1458761,
			"text": "This update's test gating status has been changed to 'ignored'.",
			"timestamp": "2020-07-15 12:33:26",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1458762,
			"text": "This update's test gating status has been changed to 'waiting'.",
			"timestamp": "2020-07-15 12:33:27",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1458810,
			"text": "This update's test gating status has been changed to 'ignored'.",
			"timestamp": "2020-07-15 13:27:47",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1459526,
			"text": "This update has been pushed to testing.",
			"timestamp": "2020-07-16 01:39:07",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1465106,
			"text": "churchyard edited this update.",
			"timestamp": "2020-07-21 14:42:55",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 1465115,
			"text": "This update has been obsoleted by [python38-3.8.5-1.fc31](https://bodhi.fedoraproject.org/updates/FEDORA-2020-bb919e575e).",
			"timestamp": "2020-07-21 14:43:44",
			"update_id": 219525,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		}
	],
	"content_type": "rpm",
	"date_modified": "2020-07-21 14:42:55",
	"date_pushed": "2020-07-16 01:38:33",
	"date_submitted": "2020-07-15 12:33:25",
	"date_testing": "2020-07-16 01:38:33",
	"meets_testing_requirements": true,
	"notes": "Security fix for CVE-2020-14422\n",
	"release": {
		"branch": "f31",
		"candidate_tag": "f31-updates-candidate",
		"composed_by_bodhi": true,
		"create_automatic_updates": false,
		"dist_tag": "f31",
		"id_prefix": "FEDORA",
		"long_name": "Fedora 31",
		"mail_template": "fedora_errata_template",
		"name": "F31",
		"override_tag": "f31-override",
		"package_manager": "dnf",
		"pending_signing_tag": "f31-signing-pending",
		"pending_stable_tag": "f31-updates-pending",
		"pending_testing_tag": "f31-updates-testing-pending",
		"stable_tag": "f31-updates",
		"state": "archived",
		"testing_repository": "updates-testing",
		"testing_tag": "f31-updates-testing",
		"version": "31"
	},
	"severity": "medium",
	"stable_days": 7,
	"stable_karma": 2,
	"status": "obsolete",
	"suggest": "unspecified",
	"test_gating_status": "ignored",
	"title": "python38-3.8.4-1.fc31",
	"type": "security",
	"url": "https://bodhi.fedoraproject.org/updates/FEDORA-2020-fe82e2a834",
	"unstable_karma": -2,
	"updateid": "FEDORA-2020-fe82e2a834",
	"user": {
		"avatar": "https://seccdn.libravatar.org/avatar/abc4a68ebd4cf08a8dad9a21fa109dad532d58a985ae3bb35ff423b49aedf37e?s=24&d=retro",
		"email": "thrnciar@redhat.com",
		"groups": [
			{
				"name": "packager"
			},
			{
				"name": "provenpackager"
			},
			{
				"name": "python-sig"
			},
			{
				"name": "ipausers"
			},
			{
				"name": "fedora-contributor"
			},
			{
				"name": "signed_fpca"
			},
			{
				"name": "fedorabugs"
			},
			{
				"name": "copr-sig"
			},
			{
				"name": "gitcopr"
			},
			{
				"name": "python-packagers-sig"
			}
		],
		"id": 5427,
		"name": "thrnciar",
		"openid": "thrnciar.id.fedoraproject.org"
	},
	"version_hash": "2acb9a6ee5b6249d04dcddc1ddb66be1d69a8c95"
}
